wemake_python_styleguide/visitors/tokenize/conditions.py

```diff
@@ -57,11 +57,36 @@ def visit_name(self, token: tokenize.TokenInfo) -> None:
         """
         self._check_implicit_elif(token)
 
+    def _check_else_belongs_to_if(self, start_index: int) -> bool:
+        previous_token = self.file_tokens[start_index - 1]
+
+        if previous_token.type != tokenize.DEDENT:
+            # This is not the first token on the line, which means that it can
+            # also be "embedded" else: x if A else B
+            return False
+
+        for token in reversed(self.file_tokens[:start_index - 1]):
+            if token.type != tokenize.NAME:
+                continue
+
+            # Here we rely upon an intuition that in Python else have to be
+            # on the same level (same indentation) as parent statement.
+            if token.start[1] == previous_token.start[1]:
+                return token.string in {'if', 'elif'}
+
+        return False  # pragma: no cover
```

Comment:
Why do we need a `no cover` here? That's a bad sign.

Reply:
I've mentioned that in the description of PR. Short version: it is very hard to craft a specific piece of code to trigger that line, and it will be incorrect Python code at the same time.

If insist, I can do that.

Reply:
Can we rewrite the logic then?

Reply:
Sure. :)